mrmustard/training/parametrized.py

```diff
@@ -54,9 +59,34 @@ def __init__(self, **kwargs):  # NOTE: only kwargs so that we can use the arg na
             bounds = kwargs.get(f"{name}_bounds", None)
             param = create_parameter(value, name, is_trainable, bounds, owner)
 
-            # dynamically assign variable as attribute of the class
+            # dynamically assign parameter as attribute of the class
             self.__dict__[name] = param
-            self.param_names[k] = name
+            self.param_names.append(name)
+
+    def param_string(self, decimals: int) -> str:
+        r"""Returns a string representation of the parameter values, separated by commas and rounded
+        to the specified number of decimals. It includes only the parameters that are not arrays
+        and not the number of modes, or other parameters that are not in principle trainable.
+        Keeps the order of the parameters as they are defined in the class constructor.
+
+        Args:
+            decimals (int): number of decimals to round to
+
+        Returns:
+            str: string representation of the parameter values
+        """
+        string = ""
+        for _, value in self.kw_parameters:
+            if math.asnumpy(value).ndim == 0:  # don't show arrays
+                string += f"{math.asnumpy(value):.{decimals}g}, "
+        return string.rstrip(", ")
+
+    @property
+    def kw_parameters(self) -> Tuple[Tuple[str, Tensor]]:
+        r"""Return a list of parameters within the Parametrized object
+        if they have been passed as keyword arguments to the class constructor.
+        """
+        return tuple((name, getattr(self, name).value) for name in self.param_names)
```

Comment:
Why using `getattr(self, name)` instead of `self.name`?

Reply:
how do you call `self.name` if you have the string `name`?